arangod/Aql/VarUsageFinder.cpp

```diff
@@ -16,6 +19,18 @@ using namespace arangodb::aql;
 //      subqueries. After that walk downwards, and process subqueries downwards
 //      when hitting them.
 
+namespace {
+
+/// @brief Merge everything from source into target. (emilib::HashSet does not
+///        have a merge() method, nor extract()).
+auto mergeHashSets(VarSet& target, VarSet&& source) {
+  for (auto varPtr : source) {
+    target.emplace(varPtr);
+  }
+  source.clear();
```

Comment:
Is this "standard" for `merge` operations?

Not only woudl I find it a bit weird that `mergeHashSets` merges the second argument into the first, but also that the second would then afterwards be empty. Maybe this is really just a naming issue

Reply:
This is meant to replace
```
target.merge(std::move(source));
```
, that's where this is coming from. However, the `emilib::HashSet` does not have `merge` implemented. I'll extend this comment and add one at the caller's site.